packages/preview/src/common/test/jsdom.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { JSDOM } from 'jsdom';
+
+const dom = new JSDOM('<!doctype html><html><body></body></html>');
```

Comment:
Should not it happen before each test and after each test global should be cleaned up?

Reply:
Thanks! I forgot about the cleanup. It's now similar to `global-jsdom` and cleanup is done in `after`, cf. https://github.com/theia-ide/theia/blob/e2bfb21b98fa714cbd1b5558ae11d39bbf4d21cb/packages/preview/src/browser/markdown/markdown-preview-handler.spec.ts#L24

Also, it's important to execute this jsdom helper before the modules are loaded which rely on certain window properties. This is why it's not working to call it in `before`.

Reply:
Ah so that might be why my tests were failing [here](https://gist.github.com/epatpol/6ede50cf5da7d08ba2050558c6880a7f) even though I used jsdom() in the `before` mocha hook?

Reply:
@epatpol please pull this code up to @theia/core/src/browser/test and move jsdom dependency to the root package

Reply:
Ok will do!